src/sentry/silo.py

```diff
@@ -107,3 +108,19 @@ def override(*args: Any, **kwargs: Any) -> Any:
 
         functools.update_wrapper(override, original_method)
         return override
+
+
+def enforce_deprecated_code_path(message: str):
+    """
+    While refactoring out some core interfaces, we want tests to alert us and make guarantee that deprecated paths
+    do not run, while at the same time not causing any outage while such deprecations roll forward into production.
+
+    A call to this function is expected to fail in tests to provide developer feedback on dead code paths (that are
+    intended to be removed) while 'only' alerting in production if these code paths get executed there.
+    """
+    import sys
+
+    if "pytest" in sys.modules:
+        assert f"This code path should be deprecated and will be removed! {message}"
+    else:
+        sentry_sdk.capture_message(f"Deprecated code path found: {message}")
```

Comment:
If one of these does hit prod, it will be high volume as Access is used on every API request. Could we 'sample' this by only emitting it <5% of the time.

Reply:
Sure, teach me about our dynamic sampling offering :)

To be clear however, I don't expect in reality that this would actually be hit at high volume.  This function validates that any tested green codepath *does not* hit these methods, meaning that if there is a leak, it's almost certainly an edge case and not most auth passes.

Reply:
How about this -- what if this is changed to only fire a capture message once per process ever?  If we see one of these, it should be immediately enough information to rollback and hunt down the traceback to resolve.

Reply:
Oh, dynamic sampling won't work here as it only works on transactions. My suggestion would be to do a basic 'sampling' using random(): 

```python
import random

if random.random() < 0.05:
    sentry_sdk.capture_message(...)
````

Capturing one message per process would work too.